src/LrtSquare.sol

```diff
@@ -80,32 +108,109 @@ contract LrtSquare is
+        swapper = __swapper;
+
+        // 1_000_000_000 = 100% for limiter
+        // Limit = 3_000_000_000 -> 300% of total suppply, refill rate -> 1_000_000 -> 0.1%, so will be refilled in 3000 sec (50 mins)
+        rateLimit = BucketLimiter.create(3_000_000_000, 1_000_000);
+        emit RateLimitCapacityUpdated(0, 3_000_000_000);
+        emit RefillRateUpdated(0, 1_000_000);
+
+        // 0.5%
+        maxAcceptableSlippageForRebalancing = 0.995 ether;
     }
 
     function mint(address to, uint256 shareToMint) external onlyGovernor {
         _mint(to, shareToMint);
     }
 
-    function registerToken(address _token) external onlyGovernor {
+    function getRateLimit() external view returns (BucketLimiter.Limit memory) {
+        return rateLimit.getCurrent();
+    } 
+
+    function whitelistRebalacingOutputToken(address _token, bool _shouldWhitelist) external onlyGovernor {
+        if (_token == address(0)) revert InvalidValue();
+        if (_shouldWhitelist) {
+            if(!isTokenRegistered(_token)) revert TokenNotRegistered();
+            if (IPriceProvider(priceProvider).getPriceInEth(_token) == 0)
+                revert PriceProviderNotConfigured();
+        }
+
+        isValidRebalanceOutputToken[_token] = _shouldWhitelist;
+        emit WhitelistRebalanceOutputToken(_token, _shouldWhitelist);
+    }
+
+    function setSwapper(address _swapper) external onlyGovernor {
+        if (_swapper == address(0)) revert InvalidValue();
+        emit SwapperSet(swapper, _swapper);
+        swapper = _swapper;
+    } 
+
+    function rebalance(
+        address _fromAsset,
+        address _toAsset,
+        uint256 _fromAssetAmount,
+        uint256 _minToAssetAmount,
+        bytes calldata _data
+    ) external onlyRebalancer {
+        if (!isTokenRegistered(_fromAsset) || !isTokenRegistered(_toAsset)) revert TokenNotRegistered();
+        // Input token may have been removed from whitelist to pause deposits for that token
+        if (!isTokenWhitelisted(_toAsset)) revert TokenNotWhitelisted();
+        if (!isValidRebalanceOutputToken[_toAsset]) revert NotAValidRebalanceOutputToken();
+        
+        uint256 vaultTotalValueBefore = getVaultTokenValuesInEth(totalSupply());
+        uint256 toAssetAmountBefore = IERC20(_toAsset).balanceOf(address(this));
+        IERC20(_fromAsset).safeTransfer(swapper, _fromAssetAmount);
+        uint256 outAmount = ISwapper(swapper).swap(_fromAsset, _toAsset, _fromAssetAmount, _minToAssetAmount, _data);
+
+        uint256 vaultTotalValueAfter = getVaultTokenValuesInEth(totalSupply());
+        uint256 toAssetAmountAfter = IERC20(_toAsset).balanceOf(address(this));
+
+        if (toAssetAmountAfter - toAssetAmountBefore < _minToAssetAmount) revert InsufficientTokensReceivedFromSwapper();
+        if (vaultTotalValueAfter < vaultTotalValueBefore) {
+            uint256 slippageApplicableInPercentage = ((vaultTotalValueBefore - vaultTotalValueAfter) * 100 ether) / vaultTotalValueBefore;
```

Comment:
why 100 ether? 1e20? can we do somethign to improve readability around `1e20` things?

Reply:
Actually it should be 100 * 1 ether because we want a percentage and also want it to be in 100 decimals

Percentage is -> ((vaultTotalValueBefore - vaultTotalValueAfter) * 100) / vaultTotalValueBefore

Then we just multiply with 1 ether